src/aap_eda/api/views/activation.py

```diff
@@ -125,6 +132,59 @@ def create(self, request):
             status=status.HTTP_201_CREATED,
         )
 
+    @extend_schema(
+        request=serializers.ActivationUpdateSerializer,
+        responses={
+            status.HTTP_200_OK: serializers.ActivationReadSerializer,
+            status.HTTP_400_BAD_REQUEST: OpenApiResponse(
+                description="Invalid data to update activation."
+            ),
+            status.HTTP_409_CONFLICT: OpenApiResponse(
+                description="The activation is not allowed to be updated."
+            ),
+        },
+    )
+    def partial_update(self, request, pk):
+        activation = self.get_object()
+        if activation.is_enabled or activation.status in [
+            ActivationStatus.STARTING,
+            ActivationStatus.STOPPING,
+            ActivationStatus.DELETING,
+            ActivationStatus.RUNNING,
+            ActivationStatus.UNRESPONSIVE,
+        ]:
+            return Response(
+                data="Activation is not in disabled mode or in stopped status",
+                status=status.HTTP_409_CONFLICT,
+            )
+        serializer = self.get_serializer(
+            instance=activation, data=request.data, partial=True
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.prepare_update(activation)
+
+        old_data = model_to_dict(activation)
+        with transaction.atomic():
+            serializer.update(activation, serializer.validated_data)
+            check_related_permissions(
+                request.user,
+                serializer.Meta.model,
+                old_data,
+                model_to_dict(activation),
```

Comment:
Are you not using the same data for the old_data for the new_data?

Reply:
the line `serializer.update(activation, serializer.validated_data)` changes activation before and after.